docs/sampler.md

```diff
@@ -0,0 +1,142 @@
+The user specifies the size range for samples to be generated, specifying the minimum and maximum sizes
+as well as the step of the sweep. The user also specifies the number of random sampling trials for each sample size.
+For each trial, the sample is evaluated by computing its Wasserstein distance (WD) from the original trace -- for each minute
+in the trace -- in two dimensions, namely the number of invocations in a minute and the amount of CPU and memory resources
+used by the functions. The latter is estimated (at the minute granularity for each function, which then are summed up)
+by multiplying the number of invocations by the average (mean) duration and by the memory footprint.
+
+Note that the latter (resource) WD metric is normalized by 1000 000 to be in the same ballpark as the former (
+invocation) WD metric. The latter metric is found empirically to be 1000 000 higher than the former because function
+invocation duration average is 1000ms, invocation count average is <10/min, and memory average is 200MB.
+
+First, the sampler derives the largest sample, for which it makes several attempts (trials) when choosing the sample
+with the smallest aggregate WD distance (mean of the invocation and resource WDs). Then, each smaller sample is derived
+from the previous, larger sample; hence guaranteeing that large samples always include smaller samples that guarantees
+monotonic load increase (in terms of resource usage) when sweeping the sample size.
+
+## Getting the traces
+
+The reference traces are stored in the `reference_traces` branch of this repo in the root folder, as a `samples.tar.gz`
+file stored in Git LFS. That archive contains both the preprocessed and sampled traces
+(day 1, 00:02:00-00:02:20, 64 trials; 100-1k functions with step 100 and 1k-44k with step 1k).
+
+```console
+python3 -m sampler sample -h
+
+usage: sample [-h] -t path -o path [-min integer] [-st integer] [-max integer] [-tr integer]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -t path, --source_trace path
+                        Path to trace to draw samples from
+  -orig path, --original_trace path
+                        Path to the Azure (or other original) trace files, required to maximize the derived sample's representativity (WD from the original trace)
+  -o path, --output path
+                        Output path for the resulting samples
+  -min integer, --min-size integer
+                        Minimum sample size (#functions).
+  -st integer, --step-size integer
+                        Step (#functions) in sample size during sampling.
+  -max integer, --max-size integer
+                        Maximum sample size (#functions).
+  -tr integer, --trial-num integer
+                        Number of sampling trials for each sample size.
+```
+
+### Plotting
+
+**Note:** Currently plotting is broken and has been disabled. (Issue 77)
```

Comment:
As well, referencing the Issue from the old sampler repo. Should we transfer issues?